packages/sign-client/src/controllers/engine.ts

```diff
@@ -2941,4 +2974,56 @@ export class Engine extends IEngine {
       }
     }
   };
+
+  private shouldSetTVF = (
+    protocolMethod: JsonRpcTypes.WcMethod,
+    params: JsonRpcTypes.RequestParams["wc_sessionRequest"],
+  ) => {
+    if (!params) return false;
+    if (protocolMethod !== "wc_sessionRequest") return false;
+    const { request, chainId } = params;
+    const namespace = parseChainId(chainId).namespace;
+    if (!(namespace in TVF_METHODS)) return false;
+    const tvfMethods = TVF_METHODS[namespace as keyof typeof TVF_METHODS];
+    if (!tvfMethods) return false;
+    return tvfMethods.includes(request.method);
+  };
+
+  private getTVFParams = (
+    id: number,
+    params: JsonRpcTypes.RequestParams["wc_sessionRequest"],
+    txHashes?: string[],
+  ) => {
+    try {
+      const tvf: RelayerTypes.ITVF = {
+        correlationId: id,
+        rpcMethods: [params.request.method],
+        chainId: params.chainId,
+        ...(this.isValidContractData(params.request.params) && {
+          // initially only get contractAddresses from EVM txs
+          contractAddresses: [params.request.params?.[0]?.to],
+        }),
+        txHashes,
+      };
+      return tvf;
+    } catch (e) {
+      this.client.logger.warn("Error getting TVF params", e);
+    }
+    return {};
```

Comment:
Curious why you're adopting this pattern of returning outside of the catch ?

Reply:
seems easier to read whats going on in the fx